### PR TITLE
Add docker as an runnable environment

### PR DESCRIPTION
**Pull request by PBXg33k**
I've migrated all my servers (and roles) on my homelab to docker and thus been porting all servers to be able to run on docker.

This PR adds docker as an additional environment.
Using docker-compose only requires one copy action and changes to the docker-compose.override.yml file to have a local running instance

### CODE REVIEW

**Comment by PBXg33k** [comment]
Adding WIP tag as i've discovered some small issues which i'm adressing

**Comment by Sn0wCrack** [comment]
That's pretty nice, I'm not a docker guy myself, but let me know once you've worked out all the kinks and I'll take a look at merging this in.

**Comment by PBXg33k** [comment]
Yes, i've got most kinks worked out locally and in my swarm. The biggest challenge was Sphins Search, mostlikely because of lack of experience.

At this moment i'm running manual tests on the docker stack, missing extensions, cronjobs not getting picked up, small stuff like that

**Comment by PBXg33k** [comment]
@Sn0wCrack 
Done
- [x] Import SQL file on first run using docker-compose
- [x] Implement Linux instructions to dockerfile
- [x] Crontab in app and sphinx
- [x] Create config.json based on environment variables

Notes:
I half arsed the setup for cron in the sphinx container which will run the install with every start, which is no biggy. I didn't want to add extra dockerfiles for an external dependency.

File: lib/PHPImageWorkshop/ImageWorkshop.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public static function initFromPath($path, $fixOrientation = false)
         switch ($mimeContentType) {
             case 'jpeg':
                 $image = imageCreateFromJPEG($path);
-                if (false === ($exif = @read_exif_data($path))) {
+                if (false === ($exif = @\exif_read_data($path))) {
```
Comment (Sn0wCrack) [inline lib/PHPImageWorkshop/ImageWorkshop.php]: Can we split this part out into its own PR

File: classes/ApiHandler.php [hunk 2]
```diff
@@ -487,7 +487,7 @@ protected function sendResponse($resp)
     {
         $body = json_encode($resp);
 
-        ob_end_clean();
+        @ob_end_clean();
```
Comment (Sn0wCrack) [inline classes/ApiHandler.php]: Can we split this part out into its own PR

**Comment by Sn0wCrack** [comment]
Sweet, looks good to me.

Should be fine with the Sphinx setup there honestly so long as it's not touching the data directories you should be right.

Just a couple things:

1. Would you be able to just squash your commits to keep things a little nicer looking.

2. With the two small code changes, could you just split those out into a different PR? Once again, like to just keep things a bit better manage (which is rich coming from me with my commit history). I'll put comments on the lines I mean in your PR.

**Comment by PBXg33k** [comment]
commits squashed. Opening new PR with code changes.

**Comment by Sn0wCrack** [comment]
All merged in now 👍
